daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/PrimitivesDateTime.scala

```diff
@@ -191,3 +185,46 @@ case class ConvertTextDateTimePrim(e: ElementBase) extends ConvertTextCalendarPr
   protected override val implicitPattern = "uuuu-MM-dd'T'HH:mm:ss"
   protected override val validFormatCharacters = "adDeEFGhHkKmMsSuwWvVyXxYzZ".toSeq
 }
+
+abstract class ConvertBinaryCalendarPrimBase(e: ElementBase, guard: Boolean, lengthInBits: Long)
+  extends ConvertCalendarPrimBase(e, guard) {
+
+  lazy val epochCalendar: Calendar = {
+    val cal = Calendar.getInstance
+    val sdf = new SimpleDateFormat("uuuu-MM-dd'T'HH:mm:ss", ULocale.ENGLISH)
```

Comment:
For the locale parameter, should this use dfdl:calendarLanguage instead of ULocale.ENGLISH?

Unrelated to this bug, this might be the problem with DAFFODIL-1945--it looks like whenever we create a SimpleDateFormat we do not specify a locale in the constructor, so it's probably using the default locale. I *think* we want to always provide dfdl:calenarLanguage. I'll update that bug with this information.

Reply:
I updated this to use calendarLanguage but then reading comments below about calendarTimeZone have me wondering if this is correct since calendarLanguage is in the same table that specifies the properties are for text or binary with a packed decimal representation, not binarySeconds or binaryMilliseconds. I would think using calendarLanguage here is still better but want to check.

Reply:
I think you are doing this right. Calendar language won't matter if these objects end up being used for binary calendars. For text calendars, calendar language should be used.